### PR TITLE
Fix uniqueIds from incrementing on toggle

### LINKED ISSUE #3598: [Bug] Toggle, aria-controls uniqueidentifer will not increment by 1 on every toggle click
**User Story:**
As Spark we want to fix React Toggle and SprkMastheadAccordionItem so that aria-controls uniqueidentifer will no longer increment by 1 on every toggle click.

**Notes:**
Requestor, SME: Katie 10/13

**Current State:**
The aria-controls uniqueidentifer is being incremented by 1 on every toggle click. This was found while working on issue #3295

**AC:**
- Update will be made to React Toggle 
- Update will be made to SprkMastheadAccordionItem
- aria-controls uniqueidentifer will not increment by 1 on every toggle click

### PR DESCRIPTION
**Pull request by guern1kn**
## What does this PR do?
Updates the uniqueIds in SprkToggle and SprkMastheadAccordionItem to be unique if there are multiple on the same page, but not increment on toggle.

### Associated Issue
Fixes #3598 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR then please remove it.

### Code
 - [x] Build Component in React
 - [x] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [x] Apple Safari
  - [ ] Apple Safari (Mobile)

### Screenshots
Example of multiple toggles on the same page with different ids.
image

### CODE REVIEW

File: react/src/components/masthead/components/SprkMastheadAccordionItem/SprkMastheadAccordionItem.test.js [hunk 1]
```diff
@@ -161,31 +157,34 @@ describe('SprkMastheadAccordionItem:', () => {
...
     const menuItemContent = wrapper.find('.sprk-c-MastheadAccordion__details');
     const menuItemContentId = menuItemContent.getDOMNode().getAttribute('id');
-    const menuItemToggleAriaControls = menuItemToggle.getDOMNode().getAttribute('aria-controls');
+    const menuItemToggleAriaControls = menuItemToggle
+      .getDOMNode()
+      .getAttribute('aria-controls');
 
     expect(menuItemContentId).toEqual(testItemId);
     expect(menuItemToggleAriaControls).toEqual(testItemId);
   });
```
Comment (bobwise) [inline react/src/components/masthead/components/SprkMastheadAccordionItem/SprkMastheadAccordionItem.test.js:189]: Great job on this fix! It looks like we already have tests to make sure a provided `id` is not overwritten by a calculated `id` - can we get a new test to make sure the id doesn't change when we toggle?

File: react/src/components/toggle/SprkToggle.test.js [hunk 2]
```diff
@@ -165,7 +165,10 @@ describe('SprkToggle:', () => {
...
   are set`, () => {
     const wrapper = mount(
-      <SprkToggle triggerText="Toggle title" triggerTextAdditionalClasses="test">
+      <SprkToggle
+        triggerText="Toggle title"
+        triggerTextAdditionalClasses="test"
+      >
         Body text
       </SprkToggle>,
     );
```
Comment (bobwise) [inline react/src/components/toggle/SprkToggle.test.js:174]: Same feedback as Masthead - let's make sure to add a new test to verify that the ID doesn't change on toggle.

**Review by bobwise — CHANGES_REQUESTED** [review]
Great job on this bug fix! I just left a couple comments.

File: react/src/components/toggle/SprkToggle.test.js [hunk 3]
```diff
@@ -113,6 +113,33 @@ describe('SprkToggle:', () => {
     expect(wrapper.find(`div#${toggleContentId}`).length).toBe(1);
   });
 
+  it('should not change id when toggled', () => {
+    const wrapper = mount(
+      <SprkToggle triggerText="Toggle title">Body text</SprkToggle>,
+    );
+    const ButtonAriaControls = wrapper.find('button').prop('aria-controls');
```
Comment (bobwise) [inline react/src/components/toggle/SprkToggle.test.js:120]:
```suggestion
    const buttonAriaControls = wrapper.find('button').prop('aria-controls');
```

File: react/src/components/toggle/SprkToggle.test.js [hunk 4]
```diff
@@ -113,6 +113,33 @@ describe('SprkToggle:', () => {
...
+    const toggleContentId = wrapper
+      .find('.sprk-c-Toggle__content')
+      .at(1)
+      .prop('id');
+
+    expect(toggleContentId).toMatch(/sprk_toggle_content_\d/);
+    expect(toggleContentId).toEqual(ButtonAriaControls);
```
Comment (bobwise) [inline react/src/components/toggle/SprkToggle.test.js:127]:
```suggestion
    expect(toggleContentId).toEqual(buttonAriaControls);
```

File: react/src/components/toggle/SprkToggle.test.js [hunk 5]
```diff
@@ -113,6 +113,33 @@ describe('SprkToggle:', () => {
...
+    wrapper.find('button').simulate('click');
+
+    const postClickToggleContentId = wrapper
+      .find('.sprk-c-Toggle__content')
+      .at(1)
+      .prop('id');
+
+    expect(wrapper.state().isOpen).toBe(true);
+    expect(toggleContentId).toMatch(postClickToggleContentId);
+    expect(ButtonAriaControls).toEqual(postClickToggleContentId);
```
Comment (bobwise) [inline react/src/components/toggle/SprkToggle.test.js:140]:
```suggestion
    expect(buttonAriaControls).toEqual(postClickToggleContentId);
```

**Review by bobwise — APPROVED** [review]

File: react/src/components/toggle/SprkToggle.test.js (hunk 3 above)
Comment (bobwise) [reply react/src/components/toggle/SprkToggle.test.js:120]: I think this should be lowercase.

**Review by RCopeland — APPROVED** [review]
lgtm

**Pull request merged**